video/src/ai/eto/rikai/sql/spark/datasources/FrameIdFilterV1.scala

```diff
@@ -0,0 +1,54 @@
+package ai.eto.rikai.sql.spark.datasources
+
+import org.apache.spark.sql.sources.{
+  EqualNullSafe,
+  EqualTo,
+  Filter,
+  GreaterThan,
+  GreaterThanOrEqual,
+  LessThan,
+  LessThanOrEqual
+}
+
+/** Push down by frame_id using lowest and highest
+  * @param lowest
+  * @param highest
+  */
+case class FrameIdFilterV1(lowest: Option[Long], highest: Option[Long])
+
+object FrameIdFilterV1 {
+  val DEFAULT_LOWEST_ID = 1L
+  val DEFAULT_HIGHEST_ID = Long.MaxValue
+
+  def apply(filters: Seq[Filter]): FrameIdFilterV1 = {
```

Comment:
Would some scenario like filters equal to `Seq(GreaterThan(_, 10), GreaterThan(_, 20))` happen? If it would happen, the lowest value would be 10, but I think it should be 20?

Reply:
I added a test in my local branch
```
  test("between 10 and 13 but with teasing") {
    val (rows1, rows2) = generateRows("frame_id > 10 and frame_id > 1 and frame_id < 13")
    assertRows(rows1, rows2, 2)
  }
```
It passed, but from the log I added, I saw
```
filters: WrappedArray(IsNotNull(frame_id), GreaterThan(frame_id,10), GreaterThan(frame_id,1), LessThan(frame_id,13))
res: FrameIdFilterV1(Some(1),Some(13))
```
It's weird.

Reply:
FrameIdFilterV1 is very naive. We might merge "frame_id > 10 and frame_id > 1" into "frame_id > 10" later.

Reply:
We'd better leave some comments at the beginning of the algorithm to explain it.